### PR TITLE
hibernate

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

File: src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]:
```suggestion
 session.save(movie);
```

Comment (ShemsheyMaxim) [reply src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]: about this write in checklist.

File: src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]: you can delete this

File: src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                session.close();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("select Movie from Movie", Movie.class).getResultList();
```
Comment (ShemsheyMaxim) [inline src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]:
```suggestion
            return session.createQuery("FROM Movie", Movie.class).getResultList();
```

File: src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessException("Can`t insert Movie entity", e);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessException("Can`t insert movie " + movie + , e);
```

Comment (ShemsheyMaxim) [reply src/main/java/com/hibernate/dao/impl/MovieDaoImpl.java]: you can do exception more informative

File: src/main/java/com/hibernate/exception/DataProcessException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.hibernate.exception;
+
+public class DataProcessException extends RuntimeException {
+    public DataProcessException(String message, Exception e) {
+        super(message);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/hibernate/exception/DataProcessException.java]:
```suggestion
    public DataProcessException(String message, Throwable cause) {
        super(message, cause);
```

**Review by ShemsheyMaxim — APPROVED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,23 +1,3 @@
...
-*.ear
-*.zip
-*.tar.gz
-*.rar
-
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+.idea/*
+*.iml
+target/*
```
Comment (Sofasmile) [inline .gitignore]: add empty line at the end of the file

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (Sofasmile) [inline .travis.yml]: same here

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+                <version>3.8.0</version>
+                <configuration>
+                    <source>9</source>
+                    <target>9</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (Sofasmile) [inline pom.xml]: same

File: src/main/java/com/hibernate/model/CinemaHall.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package com.hibernate.model;
+
+public class CinemaHall {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/CinemaHall.java]: this class shouldn't be in the PR

you car remove all redundant files from git control using the command:  `git rm --cached filename`
after that, you should commit all changes and push them to the PR. 

Then the files will stay locally in the project, but they will not be in PR

File: src/main/java/com/hibernate/model/Movie.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+@Entity
+public class Movie {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String description;
+
+    public long getId() {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/Movie.java]:
```suggestion
    public Long getId() {
```

File: src/main/java/com/hibernate/model/Movie.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        return id;
+    }
+
+    public void setId(long id) {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/Movie.java]:
```suggestion
    public void setId(Long id) {
```

File: src/main/java/com/hibernate/model/MovieSession.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
+package com.hibernate.model;
+
+import java.time.LocalDateTime;
+
+public class MovieSession {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/MovieSession.java]: the same

File: src/main/java/com/hibernate/model/Order.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
+package com.hibernate.model;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class Order {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/Order.java]: the same

File: src/main/java/com/hibernate/model/ShoppingCard.java [hunk 14]
```diff
@@ -0,0 +1,36 @@
+package com.hibernate.model;
+
+import java.util.List;
+
+public class ShoppingCard {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/ShoppingCard.java]: the same

File: src/main/java/com/hibernate/model/Ticket.java [hunk 15]
```diff
@@ -0,0 +1,34 @@
+package com.hibernate.model;
+
+public class Ticket {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/Ticket.java]: the same

File: src/main/java/com/hibernate/model/User.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+package com.hibernate.model;
+
+public class User {
```
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/User.java]: the same

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/model/Movie.java (hunk 10 above)
Comment (Sofasmile) [reply src/main/java/com/hibernate/model/Movie.java]: how about this comment?

File: src/main/java/com/hibernate/model/User.java (hunk 16 above)
Comment (Sofasmile) [inline src/main/java/com/hibernate/model/User.java]: let's remove User too

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/model/Movie.java (hunk 10 above)
Comment (AndreiYazadzhi) [reply src/main/java/com/hibernate/model/Movie.java:16]: image

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**